libnymea-zigbee/zcl/zigbeecluster.cpp

```diff
@@ -208,7 +208,7 @@ ZigbeeClusterReply *ZigbeeCluster::createClusterReply(const ZigbeeNetworkRequest
     connect(zclReply, &ZigbeeClusterReply::finished, this, [this, zclReply](){
         zclReply->deleteLater();
         m_pendingReplies.remove(zclReply->transactionSequenceNumber());
-    }, Qt::QueuedConnection);
```

Comment:
are you really sure about this one?
There must have been a reason for this like (reply being finished right after creation due to hardware issue or something). This might break the some logic relaying on: "even if a reply is finished right after creating it, I still need to process it (in the m_pendingReplies) and stuff...

Reply:
Yes... in fact, I've had crashes because of this... Because if we connect to finished() in a queued slot, it could be (and it did happen with the timeout) that the zclReply has been deleted in between the scheduling of this slot and it actually calling it. In other words, even if the connect is on the zclReply object, if using a QueuedConnection you can't be sure that it still exists by the time the slot is called.

Have been observing this for a while now and it's fine.